Var-file causes errors for terraform validate after 0.15.0

According to the [0.15.0 release notes](https://github.com/hashicorp/terraform/releases/tag/v0.15.0) -var-file is no longer valid for terraform validate. After 0.12 this option was ignored but did not cause an error. In 0.15, providing this causes the following error.

```text
2021-04-15T16:26:21.3455555Z ##[section]Starting: terraform validate
2021-04-15T16:26:21.3466179Z ==============================================================================
2021-04-15T16:26:21.3466829Z Task         : Terraform CLI
2021-04-15T16:26:21.3467243Z Description  : Execute terraform cli commands
2021-04-15T16:26:21.3467535Z Version      : 0.6.20
2021-04-15T16:26:21.3467906Z Author       : Charles Zipp
2021-04-15T16:26:21.3468291Z Help         : 
2021-04-15T16:26:21.3468760Z ==============================================================================
2021-04-15T16:26:21.8462145Z [command]/opt/hostedtoolcache/terraform/0.15.0/x64/terraform version
2021-04-15T16:26:21.9253003Z Terraform v0.15.0
2021-04-15T16:26:21.9253759Z on linux_amd64
2021-04-15T16:26:21.9254313Z + provider registry.terraform.io/hashicorp/azurerm v2.55.0
2021-04-15T16:26:21.9314643Z 
2021-04-15T16:26:22.5203959Z [command]/opt/hostedtoolcache/terraform/0.15.0/x64/terraform validate -var-file=/home/vsts/work/_temp/default.vars
2021-04-15T16:26:22.5772067Z [31m╷[0m[0m
2021-04-15T16:26:22.5773901Z [31m│[0m [0m[1m[31mError: [0m[0m[1mFailed to parse command-line flags[0m
2021-04-15T16:26:22.5775355Z [31m│[0m [0m
2021-04-15T16:26:22.5776511Z [31m│[0m [0m[0mflag provided but not defined: -var-file
2021-04-15T16:26:22.5777777Z [31m╵[0m[0m
2021-04-15T16:26:22.5778238Z 
2021-04-15T16:26:22.5779043Z For more help on using this command, run:
2021-04-15T16:26:22.5779999Z   terraform validate -help
2021-04-15T16:26:22.5875811Z 
2021-04-15T16:26:22.5917453Z ##[error]Terraform command 'validate' failed with exit code '1'.
2021-04-15T16:26:22.5935703Z ##[error][31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mFailed to parse command-line flags[0m
[31m│[0m [0m
[31m│[0m [0m[0mflag provided but not defined: -var-file
[31m╵[0m[0m

For more help on using this command, run:
  terraform validate -help

2021-04-15T16:26:23.2944148Z ##[section]Finishing: terraform validate
```

Workaround is to remove the `secureVarsFile` input from the terraform validate command. Also ensure `-var-file` is not provided in the `commandOptions` input.

--- a/tasks/terraform-cli/src/commands/tf-validate.ts
+++ b/tasks/terraform-cli/src/commands/tf-validate.ts
@@ -1,7 +1,7 @@
 import { ICommand, CommandResponse } from ".";
 import { IRunner } from "../runners";
 import { ITaskAgent } from "../task-agent";
-import { RunWithTerraform } from "../runners/builders";
+import { RunnerOptionsBuilder, RunWithTerraform } from "../runners/builders";
 import { ITaskContext } from "../context";
 
 export class TerraformValidate implements ICommand {
@@ -14,10 +14,13 @@ export class TerraformValidate implements ICommand {
     }
 
     async exec(ctx: ITaskContext): Promise<CommandResponse> {
-        const options = await new RunWithTerraform(ctx)
-            .withSecureVarFile(this.taskAgent, ctx.secureVarsFileId, ctx.secureVarsFileName)
-            .withCommandOptions(ctx.commandOptions)
-            .build();
+        let builder: RunnerOptionsBuilder = new RunWithTerraform(ctx);
+
+        if(!ctx.terraformVersionMinor || ctx.terraformVersionMinor < 15){
+            builder = builder.withSecureVarFile(this.taskAgent, ctx.secureVarsFileId, ctx.secureVarsFileName)
+        }
+        builder = builder.withCommandOptions(ctx.commandOptions);
+        const options = await builder.build();
         const result = await this.runner.exec(options);
         return result.toCommandResponse();
     }

